lib/targets/cuda/device.cpp

```diff
@@ -13,6 +13,19 @@ static const int Nstream = 9;
 #define CHECK_CUDA_ERROR(func)                                                                                         \
   target::cuda::set_runtime_error(func, #func, __func__, __FILE__, __STRINGIFY__(__LINE__));
 
+#define NVML_CHECK(func)						\
+  {									\
+    nvmlReturn_t ret = func;						\
+    if (ret == NVML_ERROR_NOT_SUPPORTED) {				\
+      static int i=0;							\
+      if (i==0) printf("%s not supported on this GPU\n", #func);	\
+      i++;								\
+    } else if (ret != NVML_SUCCESS) {					\
+      printf("Error %s returns %s\n", #func, nvmlErrorString(ret));	\
+      exit(-1);								\
+    }									\
+  }
+
```

Comment:
Needs a clang-format. Also should we just turn off monitor entirely in the `NVML_ERROR_NOT_SUPPORTED` case?

Reply:
I've update the code to issue a `warningQuda`, but I think we can just let the monitoring continue, as this may only affect a subset of the measurements.  In practice, I don't think this will ever be a real issue as I think this only affect older GeForce cards, and all modern GPUs support all these queries.

Reply:
[c52fac3](https://github.com/lattice/quda/pull/1479/commits/c52fac3e6e75dc3479d0c01935e714cc04c91802)